test/system_SUITE.erl

```diff
@@ -0,0 +1,449 @@
+                                                               %% disable ssl session caching
+                                                               %% as this ensures the cert
+                                                               %% will be re-verified by the
+                                                               %% server
+                                                               {reuse_sessions, false},
+                                                               {key, KeyFirst} ]}),
+
+    %% And: the updated certificate should allow the user to connect
+    {ok, Con2} =
+     amqp_connection:start(#amqp_params_network{host = "127.0.0.1",
+                                                port = Port,
+                                                ssl_options = [{cert, CertUpdated},
+                                                               {reuse_sessions, false},
+                                                               {key, KeyUpdated} ]}),
+    ok = amqp_connection:close(Con2),
+    %% Clean: server TLS/TCP.
+    ok = delete("bart.pem"),
+    ok = rabbit_networking:stop_tcp_listener(Port).
+
+
+ensure_configuration_using_binary_strings_is_handled(Config) ->
+    ok = rabbit_ct_broker_helpers:rpc(Config, 0,
+           ?MODULE, ensure_configuration_using_binary_strings_is_handled1, []). 
+
+ensure_configuration_using_binary_strings_is_handled1() ->
+    ok = change_configuration(rabbitmq_trust_store, [{directory, list_to_binary(friendlies())},
+                                                    {refresh_interval,
+                                                        {seconds, interval()}}]).
+
+%% Test Constants
+
+port() -> 4096.
+
+data_directory() ->
+    Path = os:getenv("TMPDIR"),
+    %% Ensure TMPDIR exists
+    true = (false =/= Path),
+    Path.
+
+friendlies() ->
+    filename:join([data_directory(), "friendlies"]).
+
+build_directory_tree(Path) ->
+    ok = filelib:ensure_dir(Path),
+    file:make_dir(Path).
+
+force_delete_entire_directory(Path) ->
+    rabbit_file:recursive_delete([Path]).
+
+interval() ->
+    1.
+
+wait_for_file_system_time() ->
+    timer:sleep(timer:seconds(1)).
+
+wait_for_trust_store_refresh() ->
+    timer:sleep(2 * timer:seconds(interval())).
+
+cfg() ->
+    {ok, Cfg} = application:get_env(rabbit, ssl_options),
+    Cfg.
+
+%% Ancillary
+
+chain(Issuer) ->
+    %% Theses are DER encoded.
+    {Certificate, {Kind, Key, _}} = erl_make_certs:make_cert([{key, dsa}, {issuer, Issuer}]),
+    {Certificate, {Kind, Key}}.
+
+change_configuration(App, Props) ->
+    ok = application:stop(App),
+    ok = change_cfg(App, Props),
+    application:start(App).
+
+change_cfg(_, []) ->
+    ok;
+change_cfg(App, [{Name,Value}|Rest]) ->
+    ok = application:set_env(App, Name, Value),
+    change_cfg(App, Rest).
+
+whitelist(Path, Filename, Certificate, {A, B} = _Key) ->
+    ok = erl_make_certs:write_pem(Path, Filename, {Certificate, {A, B, not_encrypted}}),
```

Comment:
Instead of generating new certs for each run, can you add them permanently to the testsuite?

You can store testsuite-specific data in `test/system_SUITE_data` and retrive that path using `?config(data_dir, Config)`.